dbms/src/Server/Server.cpp

```diff
@@ -278,6 +299,10 @@ struct TiFlashProxyConfig
                 args_map[engine_store_advertise_address] = args_map[engine_store_address];
 
             args_map[engine_label] = getProxyLabelByDisaggregatedMode(disaggregated_mode);
+            if (disaggregated_mode != DisaggregatedMode::Compute && hasS3Config(config))
```

Comment:
How about simply checking storage.s3.bucket? It is more unified with the rest of the code, act in the same way as how they check S3 is enabled.

Reply:
+1
After merging the master to this branch, the StorageConfig will be parsed before ProxyConfig

https://github.com/pingcap/tiflash/blob/d51c5dfa11c5697320a10087915f2e8d9ea82455/dbms/src/Server/Server.cpp#L874-L892

Reply:
> How about simply checking storage.s3.bucket? It is more unified with the rest of the code, act in the same way as how they check S3 is enabled.

Just check disaggregated_mode to avoid compute node to set `engine_role`:`write`.